File: src/main/java/ar/edu/unlam/tallerweb1/infrastructure/utils/JdbcUtil.java
package ar.edu.unlam.tallerweb1.infrastructure.utils;

import java.sql.*;
import java.time.LocalDateTime;

public class JdbcUtil {

    public static PreparedStatement prepareStatement(Connection connection, String sql) {
        try {
            return connection.prepareStatement(sql);
        } catch (SQLException e) {
            throw new RuntimeException(e);
        }
    }

    public static void setLong(int index, Long value, PreparedStatement preparedStatement) {
        try {
            preparedStatement.setLong(index, value);
        } catch (SQLException e) {
            throw new RuntimeException(e);
        }
    }

    public static void setDouble(int index, Double value, PreparedStatement preparedStatement) {
        try {
            preparedStatement.setDouble(index, value);
        } catch (SQLException e) {
            throw new RuntimeException(e);
        }
    }

    public static void setString(int index, String value, PreparedStatement preparedStatement) {
        try {
            preparedStatement.setString(index, value);
        } catch (SQLException e) {
            throw new RuntimeException(e);
        }
    }

    public static void setDate(int index, LocalDateTime value, PreparedStatement preparedStatement) {
        try {
            preparedStatement.setDate(index, Date.valueOf(value.toLocalDate()));
        } catch (SQLException e) {
            throw new RuntimeException(e);
        }
    }

    public static ResultSet executeQuery(PreparedStatement preparedStatement) {
        try {
            return preparedStatement.executeQuery();
        } catch (SQLException e) {
            throw new RuntimeException(e);
        }
    }

    public static int execute(PreparedStatement preparedStatement) {
        try {
            return preparedStatement.executeUpdate();
        } catch (SQLException e) {
            throw new RuntimeException(e);
        }
    }

    public static boolean next(ResultSet resultSet) {
        try {
            return resultSet.next();
        } catch (SQLException e) {
            throw new RuntimeException(e);
        }
    }

    public static long getLong(String column, ResultSet resultSet) {
        try {
            return resultSet.getLong(column);
        } catch (SQLException e) {
            throw new RuntimeException(e);
        }
    }

    public static double getDouble(String column, ResultSet resultSet) {
        try {
            return resultSet.getDouble(column);
        } catch (SQLException e) {
            throw new RuntimeException(e);
        }
    }

    public static String getString(String column, ResultSet resultSet) {
        try {
            return resultSet.getString(column);
        } catch (SQLException e) {
            throw new RuntimeException(e);
        }
    }

    public static LocalDateTime getDate(String column, ResultSet resultSet) {
        try {
            return resultSet.getTimestamp("date").toLocalDateTime();
        } catch (SQLException e) {
            throw new RuntimeException(e);
        }
    }
}

File: src/main/java/ar/edu/unlam/tallerweb1/infrastructure/utils/JdbcUtil.kt
package ar.edu.unlam.tallerweb1.infrastructure.utils

import java.sql.Connection
import java.sql.Date.valueOf
import java.sql.PreparedStatement
import java.sql.ResultSet
import java.sql.SQLException
import java.time.LocalDateTime

object JdbcUtil {
    fun prepareStatement(connection: Connection, sql: String): PreparedStatement {
        try {
            return connection.prepareStatement(sql)
        } catch (e: SQLException) {
            throw RuntimeException(e)
        }
    }

    fun executeQuery(preparedStatement: PreparedStatement): ResultSet {
        try {
            return preparedStatement.executeQuery()
        } catch (e: SQLException) {
            throw RuntimeException(e)
        }
    }

    fun execute(preparedStatement: PreparedStatement): Int {
        try {
            return preparedStatement.executeUpdate()
        } catch (e: SQLException) {
            throw RuntimeException(e)
        }
    }

    fun next(resultSet: ResultSet): Boolean {
        try {
            return resultSet.next()
        } catch (e: SQLException) {
            throw RuntimeException(e)
        }
    }

    fun getLong(column: String, resultSet: ResultSet): Long {
        try {
            return resultSet.getLong(column)
        } catch (e: SQLException) {
            throw RuntimeException(e)
        }
    }

    fun getDouble(column: String, resultSet: ResultSet): Double {
        try {
            return resultSet.getDouble(column)
        } catch (e: SQLException) {
            throw RuntimeException(e)
        }
    }

    fun getString(column: String, resultSet: ResultSet): String {
        try {
            return resultSet.getString(column)
        } catch (e: SQLException) {
            throw RuntimeException(e)
        }
    }

    fun getDate(column: String, resultSet: ResultSet): LocalDateTime {
        try {
            return resultSet.getTimestamp("date").toLocalDateTime()
        } catch (e: SQLException) {
            throw RuntimeException(e)
        }
    }
}

fun PreparedStatement.withLong(index: Int, value: Long): PreparedStatement {
    try {
        this.setLong(index, value)
    } catch (e: SQLException) {
        throw RuntimeException(e)
    }
    return this
}

fun PreparedStatement.withDouble(index: Int, value: Double): PreparedStatement {
    try {
        this.setDouble(index, value)
    } catch (e: SQLException) {
        throw RuntimeException(e)
    }
    return this
}

fun PreparedStatement.withString(index: Int, value: String): PreparedStatement {
    try {
        this.setString(index, value)
    } catch (e: SQLException) {
        throw RuntimeException(e)
    }
    return this
}

fun PreparedStatement.withDate(index: Int, value: LocalDateTime): PreparedStatement {
    try {
        this.setDate(index, valueOf(value.toLocalDate()))
    } catch (e: SQLException) {
        throw RuntimeException(e)
    }
    return this
}